docs/firmware/golioth-firmware-sdk/light-db-stream/guide-lightdb-stream-object.md

```diff
@@ -0,0 +1,235 @@
+                                            cbor_buf,
+                                            sizeof(cbor_buf),
+                                            obj_async_handler,
+                                            (void *)CBOR_ASYNC);
+```
+
+The async set functions send data to a given path, but the request is enqueued
+by the Golioth Client so that program execution is not blocked. If a callback
+function is supplied, it will be called by the Golioth Client when the set
+operation is completed. An optional callback argument can be specified, this
+data will be available in the callback function.
+
+In the above example, we tell Golioth to set the `"sensor"` path to the values
+stores in the `json_buf`/`cbor_buf` array. The length of that array is supplied,
+along with  with `async_set_handler` as the callback function. An enum value is
+used as the callback argument.
+
+### Callback Function
+
+Callback functions need to follow the type specified by the function that
+registers them. In this case a `golioth_set_cb_fn` type callback
+function must be defined:
+
+```c
+static void obj_async_handler(golioth_client_t client,
+                              const golioth_response_t* response,
+                              const char* path,
+                              void* arg) {
+    if (response->status != GOLIOTH_OK) {
+        // The set operation failed.
+        return;
+    }
+
+    // The set operation was successful!
+    switch((enum obj_type)arg) {
+        case JSON_ASYNC:
+            printf("JSON async set successful!\n");
+            break;
+        case CBOR_ASYNC:
+            printf("CBOR async set successful!\n");
+            break;
+        default:
+            printf("Async set successful, but arg value unrecognized\n");
+    }
+    return;
+}
+```
+
+The response is passed to the callback function as a `golioth_response_t` struct
+that includes `status`. It is recommended that callbacks test for the
+`GOLIOTH_OK` status which indicates a successful operation.
+
+The same callback may be used by multiple async functions. In this example, the
+`arg` value is used to determine which function led to this callback so that a
+different message may be printed.
+
+## Custom Timestamps
+
+As a convenience, data received without a timestamp will be automatically
+assigned one. However, in cases where you are caching sensor readings, you may
+send custom timestamps along with your data. Golioth will recognize and use the
+custom timestamp when storing the data.
+
+```c
+/* Create a valid JSON string that includes a timestamp */
+char json_buf[] = "{\"name\":\"Golioth\","
+                  "\"active\":true,"
+                  "\"data\":{\"reading\":1.337,\"calib\":83},"
+                  "\"time\":\"2023-08-09T08:27:49-05:00\"}";
+```
+
+Format your custom time as either Unix time or ISO 8601 format, and add it to
+your data object using any of the special key names: `t`, `ts`, `time`. The
```

Comment:
> add it to your data object

`data`  is a field in the objet, so using "data object" here could be ambiguous - without looking at the example, it's not clear if the `time` field should be at the root of the object or nested under `data`.

Reply:
I see your point on the "data object" language.

On the other question, I don't know if `time` needs to be a root value or can be present anywhere in the payload.